src/main/java/org/trips/service_framework/heplers/CmsHelper.java

```diff
@@ -0,0 +1,87 @@
+                .build();
+    }
+
+    public Map<String, Object> createRequestMap(String name, String value, String operator, boolean isAttribute) {
+        Map<String, Object> requestMap = new HashMap<>();
+        requestMap.put("name", name);
+        requestMap.put("value", CmsUtils.sanitizeInput(name, value));
+        requestMap.put("operator", operator);
+        requestMap.put("isAttribute", isAttribute);
+
+        return requestMap;
+    }
+
+    public Map<String, Object> getSearchQueryFromSkuCode(String code) {
+        Map<String, Object> requestMap = createRequestMap("code", code, "EQ", false);
+        return Map.of("searchQuery", Map.of("filters", List.of(requestMap)));
+    }
+
+    public Map<String, Object> getSearchQueryFromAttributes(SkuAttributes attributes) {
+        List<Map<String, Object>> attributeList = new ArrayList<>();
+        attributeList.add(createRequestMap("species", attributes.getSpecies(), "EQ", true));
+        attributeList.add(createRequestMap("product_type", attributes.getProductType(), "EQ", true));
+        attributeList.add(createRequestMap("spec", attributes.getSpec(), "EQ", true));
+        attributeList.add(createRequestMap("catch_type", attributes.getCatchType(), "EQ", true));
+        attributeList.add(createRequestMap("freezing_method", attributes.getFreezingMethod(), "EQ", true));
+        attributeList.add(createRequestMap("packing", attributes.getPacking(), "EQ", true));
+        attributeList.add(createRequestMap("glazing_percentage", attributes.getGlazingPercentage(), "EQ", true));
+        attributeList.add(createRequestMap("unit_weight", attributes.getUnitWeight(), "EQ", true));
+        attributeList.add(createRequestMap("quantity_per_unit", attributes.getQuantityPerUnit(), "EQ", true));
+        attributeList.add(createRequestMap("unit_per_carton", attributes.getUnitPerCarton(), "EQ", true));
+        attributeList.add(createRequestMap("treatment", attributes.getTreatment(), "EQ", true));
+        attributeList.add(createRequestMap("grade", attributes.getGrade(), "EQ", true));
+        attributeList.add(createRequestMap("quality", attributes.getQuality(), "EQ", true));
+
+        return Map.of("searchQuery", Map.of("filters", attributeList));
+    }
+
+    public String constructSkuName(List<Map<String, String>> attributes) {
+        Map<String, String> attributeMap = new HashMap<>();
+        attributes.forEach(x -> {
+            String name = x.get("name");
+            String value = x.get("value");
+            attributeMap.put(name, Objects.equals(value, CmsConstants.NOT_APPLICABLE) ? "" : value.trim());
+        });
+
+        String quantityPerUnit = attributeMap.getOrDefault("quantity_per_unit", "");
+        String unitPerCarton = attributeMap.getOrDefault("unit_per_carton", "");
+        String unitData = (quantityPerUnit.isEmpty() || unitPerCarton.isEmpty()) ? "" : String.format("%sx%s", unitPerCarton, quantityPerUnit);
```

Comment:
If I recall correctly, if both `quantityPerUnit` and `unitPerCarton` are not present we use `unitWeight`. Is that handling not present? @hardikph check this out from the business logic side once

Reply:
@hardikph check from omnitrix. It has the most recent updated cms interaction.